### ISSUE TITLE
Expose contextual descriptor's fallback descriptor

### ISSUE THREAD

**Issue by rnett** [Kotlin/kotlinx.serialization#2611]
**What is your use-case and why do you need this feature?**

If a type is annotated with `@Contextual`, but there is no serializer in context, it will use the type's own serializer (i.e. plugin-generated) if it has one.  This is contained in the `ContextualSerializer.fallbackSerializer` property, which is private.  I'm trying to do some schema introspection (based on descriptors) and this is making it impossible to have my introspected schema match the actual behavior, since I can't identify the fallback serializer used (I'm doing this in the context of a serializers module, so resolving the serializer itself is not an issue).

**Describe the solution you'd like**

Make `SerializersModule.getContextualDescriptor` return the fallback serializer's descriptor if the fallback serializer would be used (or add an overload that does this).

It's simple enough to implement, if you approve the idea I'm happy to make a PR.

**Comment by pdvrieze** [comment]
@sandwwraith @rnett I would want this for XML too.

Btw. there is a workaround way to "get" the serializer from the contextual serializer by calling deserialize with a special decoder that has no module and will record the serializer provided - it can then throw an exception instead of actually deserializing (from nothing). It would be good to get this information in a better way (especially when it would need nested "deserialization").

As an aside, there is a similar issue with `NullableSerializer`, although the serial descriptor is normally detailed enough.

**Comment by rnett** [comment]
> As an aside, there is a similar issue with NullableSerializer, although the serial descriptor is normally detailed enough.

Yeah I've run into this too, especially when trying to use them as keys for caching.
